File: checkers/sol_checker.py
import asyncio

class SolChecker:
    def __init__(self, rpc_manager):
        self.rpc_manager = rpc_manager

    async def check(self, session, address):
        for _ in range(len(self.rpc_manager.rpcs)):
            entry = self.rpc_manager.get()

            if not entry:
                return False

            url = entry["url"]

            payload = {
                "jsonrpc": "2.0",
                "id": 1,
                "method": "getBalance",
                "params": [address]
            }

            try:
                async with session.post(url, json=payload, timeout=10) as r:
                    if r.status != 200:
                        continue

                    data = await r.json()

                    if "result" in data and "value" in data["result"]:
                        balance = data["result"]["value"]
                        return balance > 0

            except:
                continue

        return False

File: checkers/solChecker.js
const { postJson } = require("../core/http");

class SolChecker {
  constructor(rpcManager) {
    this.rpcManager = rpcManager;
  }

  async check(address) {
    for (let i = 0; i < this.rpcManager.rpcs.length; i += 1) {
      const entry = this.rpcManager.get();
      if (!entry) return false;

      const data = await postJson(entry.url, {
        jsonrpc: "2.0",
        id: 1,
        method: "getBalance",
        params: [address]
      });

      if (typeof data?.result?.value !== "number") {
        this.rpcManager.markFail(entry);
        continue;
      }

      this.rpcManager.markSuccess(entry);
      return data.result.value > 0;
    }

    return false;
  }
}

module.exports = { SolChecker };
